apps/backend/src/services/product.service.ts

```diff
@@ -1,25 +1,106 @@
 import { Product } from '../data/models/product';
 import { connection } from '../main';
 
-const getAllProducts = async (menuId: string, page: number) => {
+const getAllProducts = async (
+  menuId: string,
+  page: number,
+  searchByName: string,
+  sort,
+  min_price: number,
+  max_price: number
+) => {
   try {
     const perPage = 12;
 
     const builder = connection
       .getRepository(Product)
-      .createQueryBuilder('product')
-      .where('product.menuId = :menuId', { menuId });
+      .createQueryBuilder('product');
+
+    const minAndMaxValue = connection
+      .getRepository(Product)
+      .createQueryBuilder('product');
+
+    let products;
+
+    if (page) {
+      products = await builder.where('product.menuId = :menuId', { menuId });
```

Comment:
`builder.where('product.menuId = :menuId', { menuId });` returns an updated `builder`, not products, because you did not write `getMany` here. So it's still the builder. You have to reassign the builder here
`builder = await builder.where('product.menuId = :menuId', { menuId });`